packages/bus-query-operation/lib/Bindings.ts

```diff
@@ -1,38 +1,18 @@
-import type { AsyncIterator } from 'asynciterator';
+import type { IBindings } from '@comunica/types';
 import { Map } from 'immutable';
 import type * as RDF from 'rdf-js';
 import { termToString } from 'rdf-string';
 import type { Algebra, Factory } from 'sparqlalgebrajs';
 import { Util } from 'sparqlalgebrajs';
 
-/**
- * A stream of bindings.
- *
- * Next to the list of available variables,
- * an optional metadata hash can be present.
- *
- * @see Bindings
- */
-export type BindingsStream = AsyncIterator<Bindings>;
```

Comment:
Also re-export this and following.

Reply:
Will do.

Reply:
Further to my comments on Gitter, I gave up and simply re-added and re-exported the old `Bindings` and `BindingsStream` types while using the new `TBindings` and `TBindingsStream` ones throughout the module. They are effectively identical and the TS compiler is able to figure out that they are interchangeable. This likely comes at a cost in terms of compilation performance but I haven't seen tangible differences in compilation times, so I went with this workaround hoping that the night will bring some new idea as to how to approach this.

Reply:
Did you experience any problems with just re-exporting those from the `types` package?
If not, I would propose to do that. Otherwise we'd have identical type and interface definitions in two different places, which will certainly cause maintainability problems in the future. (I would also expect jest's coverage requirement to trip over this)

Reply:
Yes, once I drop the `T` and try to re-export the `Bindings` type as imported from `@comunica/types`, the TS compiler starts complaining about duplicate exports. I don't really know how to get out of this. Will test some more later on.

Reply:
I might have found something that works. It's a bit ugly but it should prevent the maintainability issues you mentioned. The gist of it is as follows:

```js
import type { Bindings as _Bindings } from '@comunica/types';
export type Bindings = _Bindings;
function Bindings(hash) { /* ... */ }
```

The compiler seems happy and I've thus switched to this approach in my last commit. Done?